### ISSUE TITLE
Filename in web validator is not informative

### ISSUE THREAD

**Issue by jcpitre** [MobilityData/gtfs-validator#1539]
### Describe the bug

Bug or feature?
The web validator html report gives the filename of the dataset. e.g.:

![image](https://github.com/MobilityData/gtfs-validator/assets/106176106/992a452f-b7b2-4ac3-bfab-0758a58ed551)

It does not mean anything when using the web validator.

### Steps/Code to Reproduce

Any web validation will give the same result.

### Expected Results

Something significant, like the name of the uploaded file.

### Actual Results

![image](https://github.com/MobilityData/gtfs-validator/assets/106176106/0bd5b11a-9571-4b43-85f6-6f4435b8db2a)

### Validator version

version 4.1.1-SNAPSHOT

### Operating system

MacOS

**Comment by isabelle-dr** [comment]
Indeed! The intention was to have the name of the file here, which works properly when using the command line. 
<img width="909" alt="Screenshot 2023-07-27 at 2 51 55 PM" src="https://github.com/MobilityData/gtfs-validator/assets/63653518/120cd40c-5f5a-4fde-bd22-ce26e58495a4">

**Comment by bradyhunsaker** [comment]
The value printed in the report is the input file that was read. Based on the code, the web interface uploads a local file to the unique-but-not-meaningful name for processing, resulting in the current behavior.

To display the original filename in the report, we need the web interface to pass in the original file name somehow. The obvious way I see is to add an additional field to ValidationRunnerConfig. Maybe "originalGtfsSource"?

That value would be populated by the web interface around here:
https://github.com/MobilityData/gtfs-validator/blob/3d5ef6f31839f74fff3338c8a8d9869df88cc7c0/web/client/src/routes/%2Bpage.svelte#L219
Probably added to JobMetadata and populated by the server here:
https://github.com/MobilityData/gtfs-validator/blob/3d5ef6f31839f74fff3338c8a8d9869df88cc7c0/web/service/src/main/java/org/mobilitydata/gtfsvalidator/web/service/controller/ValidationController.java#L62
Then passed in to ValidationHandler.validateFeed():
https://github.com/MobilityData/gtfs-validator/blob/3d5ef6f31839f74fff3338c8a8d9869df88cc7c0/web/service/src/main/java/org/mobilitydata/gtfsvalidator/web/service/util/ValidationHandler.java#L33

The main cost is the new field in ValidationRunnerConfig that is just for the web interface. But it seems worth it.

If someone from MobilityData agrees, then I'll create a PR.

**Comment by jcpitre** [comment]
Great that you are willing to do the work. Go ahead.
It should also cover the case where a URL is provided. In that case the URL can be printed in the report.

**Comment by bradyhunsaker** [comment]
I've tried to start work on this issue, but I haven't yet succeeded at the first step, which is to be able to build and bring up a local instance of the web server (without any code changes).

I prefer not to go through the trouble of creating a Google Cloud instance. I prefer to just bring up the server locally. There are some instructions for this at https://github.com/MobilityData/gtfs-validator/blob/master/web/service/README.md#local-development, but I don't find them clear and I continue to get errors when attempting to run ./gradlew bootRun.

Do you know if anyone has brought up a local server successfully and can glance over the existing instructions?

**Comment by jcpitre** [comment]
@bradyhunsaker What about we do it in two parts: First you add an argument to the CLI version and I can work on using it in the web version.

**Comment by bradyhunsaker** [comment]
Yes, that sounds good! I wasn't imagining having a CLI interface, but this makes it seem it's a good idea for testing.

I'll test that part and create a PR.

If you'd like I could also create a draft PR for the web server part, even though I wouldn't test it.

**Comment by jcpitre** [comment]
The idea is that if you add to the CLI version most of what you write will be used by the Web version.
For the web server part, if you have not started I can take care of it. If you have started I will be happy to test it for you.

**Comment by emmambd** [comment]
Hi @bradyhunsaker, are you still planning to work on this? Our team's hoping to prioritize completing it in the next month.

**Comment by bradyhunsaker** [comment]
Hi Emma,

Apologies that I have neither done anything nor said anything in almost two months.
Thank you for asking.

There is no need to wait on me. If someone else is ready to do the work, go for it! That's true for me in general. I never want to block other folks, given that I can't usually commit to a schedule, at least not for work that I haven't started. 

In the future when I have indicated an interest in an issue like this and haven't been able to prioritize it within several weeks, I'll try to post a note explicitly saying so, so that people don't have to wonder.

I do intend to make some more contributions when I'm able to squeeze in some time, but I don't know how soon that will be, and I'm not emotionally attached to any particular issues.

**Comment by emmambd** [comment]
@bradyhunsaker No worries! We're always grateful for contributions and completely understand that capacity changes. Going forward, sharing when you haven't been able to work on something sounds like a great plan. Our team can work on this in the near term then instead.
